### PR TITLE
Allow load_type: DSHEETPILING_DETERMINED for UnformLoad and SurchargeLoad in DSheetPiling

### LINKED ISSUE #15: D-Sheet Piling does not allow LoadType: DSHEETPILING_DETERMINED for UniformLoad and SurchargeLoad
The UniformLoad and SurchargeLoad raise a ValidationError when the LoadType is set to DSHEETPILING_DETERMINED.
This is caused by the following:

The 'UniformLoad' object uses the 'VerificationLoadSettings' object

```
class UniformLoad(BaseDataClass):
    """Uniform Load. This option is not available for SignlePileModelType.

    Args:
        name: Name of the SurchargeLoad.
        left_load: Value of the load on the left side.
        right_load: Value of the load on the right side.
        verification_load_settings: VerificationLoadSettings.
        standard_deviation_left: Standard deviation of the distribution type of the left side.
        standard_deviation_right: Standard deviation of the distribution type of the right side.
        distribution_type_left: Distribution type of the left side.
        distribution_type_right: Distribution type of the right side.
    """

    name: constr(min_length=1, max_length=50)
    left_load: float
    right_load: float
    verification_load_settings: VerificationLoadSettings = VerificationLoadSettings()
```
The VerificationLoadSettingsobject uses the 'LoadTypeFavourableUnfavourableMoment ' enum object for load type
```
class VerificationLoadSettings(BaseDataClass):
    """
    Load class for moment loads
    These options are available only if the Verification (EC7/CUR) option is selected in the Model window for the D-SheetPiling model.
    For GEOLIB the "verification" parameter should be set to True in the function DSheetPilingModel.set_model

    Args:
        duration_type: Select the duration of load application, Permanent or Variable.
        load_type: Select the type of load, Favorable or Unfavorable.
    """

    load_type: LoadTypeFavourableUnfavourableMoment = (
        LoadTypeFavourableUnfavourableMoment.FAVOURABLE
    )
    duration_type: LoadTypePermanentVariable = LoadTypePermanentVariable.PERMANENT

```

The 'LoadTypeFavourableUnfavourableMoment' object  does not allow the DSHEETPILING_DETERMINED option.
```
class LoadTypeFavourableUnfavourableMoment(IntEnum):
    FAVOURABLE = 1
    UNFAVOURABLE = 2

```

The same holds for the SurchargeLoad. In D-Sheet Piling itself, this option is allowed for Uniform and Surcharge loads:

image

### PR DESCRIPTION
**Pull request by Mbweehuizen**
Hi all,

This is my proposition for the fix #15 
Please feel free to contact me with any questions. 

greeting,

Maarten Weehuizen

### CODE REVIEW

File: geolib/models/dsheetpiling/loads.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ class UniformLoad(BaseDataClass):
     name: constr(min_length=1, max_length=50)
     left_load: float
     right_load: float
-    verification_load_settings: VerificationLoadSettings = VerificationLoadSettings()
+    verification_load_settings: VerificationLoadSettingsHorizontalLineLoad = VerificationLoadSettingsHorizontalLineLoad()
```
Comment (carloslubbers) [inline geolib/models/dsheetpiling/loads.py:73]: The name of VerificationLoadSettingsHorizontalLineLoad now makes no sense anymore, it is applicable for horizontal line loads, uniform loads and surcharge loads. A more generic name, or a class per type would be preferable here

Comment (carloslubbers) [inline geolib/models/dsheetpiling/loads.py:73]: Is it OK that the default value of load type changes from FAVOURABLE to DSHEETPILING_DETERMINED?

**Review by carloslubbers — CHANGES_REQUESTED** [review]
Thanks for your contribution! Could you check my comments and consider making slight adaptations before merging? If you disagree with my suggestions feel free to let me know as well

File: geolib/models/dsheetpiling/loads.py (hunk 1 above)
Comment (Mbweehuizen) [reply geolib/models/dsheetpiling/loads.py]: I have renamed the VerificationLoadSettings classes to VerificationLoadSettingsLoads (for Surcharge, Uniform and HorizontalLineLoads) and VerificationLoadSettingsMomentNormalForce (for Moment and NormalForce). I have kept the original ones for backwards compatibility. If you want to have those removed, let me know. (or maybe add a deprecation error.

Comment (Mbweehuizen) [reply geolib/models/dsheetpiling/loads.py]: I think having DSHEETPILING_DETERMINED as a default is more logical then having FAVOURABLE for the case with 3 options.

Comment (carloslubbers) [reply geolib/models/dsheetpiling/loads.py]: Removing the old ones if unused should be fine, we can mention it in the release notes

**Review by carloslubbers — APPROVED** [review]

**Pull request merged**